src/main/haskell/language-kore/test/parser/Data/Kore/Step/AxiomPatternsTest.hs

```diff
@@ -29,12 +31,59 @@ axiomPatternsUnitTests :: TestTree
 axiomPatternsUnitTests =
     testGroup
         "AxiomPatterns Unit Tests"
-        [ testCase "A => B"
+        [ testCase "I1:AInt => I2:AInt"
             (assertEqual ""
-                undefined
+                (Right AxiomPattern
+                    { axiomPatternLeft = extractPurePattern varI1
+                    , axiomPatternRight = extractPurePattern varI2
+                    }
+                )
+                ( koreSentenceToAxiomPattern Object
+                $ asSentence $ axiomSentencePureToKore
+                    (axiom_
+                        (and_ top_
+                            (and_ top_
+                                (rewrites_ varI1 varI2)
+                            )
+                        )
+                    :: PureSentenceAxiom Object)
+                )
+            )
+        , testCase "\"a\" => \"b\""
+            (assertEqual ""
+                (koreFail "Unexpected non-Object pattern")
+                ( koreSentenceToAxiomPattern Object
+                $ asSentence
+                    (SentenceAxiom
+                        { sentenceAxiomPattern =
+                            asKorePattern $ RewritesPattern Rewrites
+                                { rewritesSort =
+                                    sortVariableSort (SortVariableName "s")
+                                , rewritesFirst =
+                                    asKorePattern $ StringLiteralPattern (StringLiteral "a")
```

Comment:
I don't think it matters much since this is a test, but this will not pass verification since it's not of sort #Pattern. You can leave it like this, but you could also wrap it in an 'equals', 'ceil', or one of the implicitly defined symbols that changes its sort. Your choice.